tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/FolderPickerProducer.java

```diff
@@ -0,0 +1,141 @@
+package org.sakaiproject.lessonbuildertool.tool.producers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.sakaiproject.lessonbuildertool.SimplePage;
+import org.sakaiproject.lessonbuildertool.SimplePageItem;
+import org.sakaiproject.lessonbuildertool.tool.beans.SimplePageBean;
+import org.sakaiproject.lessonbuildertool.tool.view.GeneralViewParameters;
+import org.sakaiproject.tool.cover.SessionManager;
+import uk.org.ponder.localeutil.LocaleGetter;
+import uk.org.ponder.messageutil.MessageLocator;
+import uk.org.ponder.rsf.components.UIContainer;
+import uk.org.ponder.rsf.components.UIBranchContainer;
+import uk.org.ponder.rsf.components.UIELBinding;
+import uk.org.ponder.rsf.components.UICommand;
+import uk.org.ponder.rsf.components.UIForm;
+import uk.org.ponder.rsf.components.UIInput;
+import uk.org.ponder.rsf.components.UIOutput;
+import uk.org.ponder.rsf.components.UIParameter;
+import uk.org.ponder.rsf.components.decorators.UIFreeAttributeDecorator;
+import uk.org.ponder.rsf.components.decorators.UITooltipDecorator;
+import uk.org.ponder.rsf.flow.jsfnav.NavigationCase;
+import uk.org.ponder.rsf.flow.jsfnav.NavigationCaseReporter;
+import uk.org.ponder.rsf.view.ComponentChecker;
+import uk.org.ponder.rsf.view.ViewComponentProducer;
+import uk.org.ponder.rsf.viewstate.SimpleViewParameters;
+import uk.org.ponder.rsf.viewstate.ViewParameters;
+import uk.org.ponder.rsf.viewstate.ViewParamsReporter;
+
+import javax.faces.component.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by neelam on 8/19/2015.
+ * Displays folders to user to add in Lessons tool
+ */
+public class FolderPickerProducer implements ViewComponentProducer, NavigationCaseReporter, ViewParamsReporter {
+
+    private SimplePageBean simplePageBean;
+    private ShowPageProducer showPageProducer;
+    public MessageLocator messageLocator;
+    public LocaleGetter localeGetter;
+
+    public static final String VIEW_ID = "FolderPicker";
+    public String getViewID() {
+        return VIEW_ID;
+    }
+    private static Log log = LogFactory.getLog(ShowPageProducer.class);
+
+    public void fillComponents(UIContainer tofill, ViewParameters viewparams, ComponentChecker checker) {
+        GeneralViewParameters gparams = (GeneralViewParameters) viewparams;
+
+        UIOutput.make(tofill, "html").decorate(new UIFreeAttributeDecorator("lang", localeGetter.get().getLanguage()))
+                .decorate(new UIFreeAttributeDecorator("xml:lang", localeGetter.get().getLanguage()));
+
+        if (gparams.getSendingPage() != -1) {
+            // will fail if page not in this site
+            // security then depends upon making sure that we only deal with this page
+            try {
+                simplePageBean.updatePageObject(gparams.getSendingPage());
+            } catch (Exception e) {
+                System.out.println("FolderPicker permission exception " + e);
```

Comment:
Should never write to `System.out`, if you want to record this use the logger and/or consider adding a message to the UI.